modules/aws/src/s3.client.ts

```diff
@@ -40,4 +44,60 @@ export class S3Client {
 
         return this.upload(filePath, payload, options);
     }
+
+    public async fileExists (filePath: string): Promise<boolean> {
+        return new Promise<boolean>((resolve: Function, reject: Function): void => {
+            this.connection.headObject({
+                Bucket: this.bucketName,
+                Key: this.generateObjectName(filePath),
+            }, (err: aws.AWSError, data: aws.S3.HeadObjectOutput): void => {
+                if (!err) {
+                    return resolve(!!data);
+                } else if (err.statusCode === 404) {
+                    return resolve(false);
+                }
+                reject(err);
+            });
+        });
+    }
+
+    public moveWithinBucket (sourceFilePath: string, targetFilePath: string): Promise<CopyObjectOutput> {
+        const sourceObjectName: string = this.generateObjectName(sourceFilePath);
+        return new Promise((resolve: Function, reject: Function): void => {
+            this.connection.copyObject({
+                Key: this.generateObjectName(targetFilePath),
+                CopySource: `${this.bucketName}/${sourceObjectName}`,
+                Bucket: this.bucketName,
+
+            }, (copyErr: aws.AWSError, output: CopyObjectOutput): void => {
+                if (copyErr) {
+                    return reject(copyErr);
+                }
+                this.connection.deleteObject({
+                    Key: sourceObjectName,
+                    Bucket: this.bucketName,
+                }, (deleteErr: aws.AWSError) => {
+                    if (deleteErr) {
+                        return reject(deleteErr);
```

Comment:
shouldn't we create smth like numerical error?
Afterwards we could have in the usage:

```
if error = 100
    log(failed to copy)
else if error = 200
    retryObjectDelete(...) or log(item copied but old file stayed, delete it manually)
```


In the end, in our scenario, we could with error 200 continue without an error